packages/cli/src/controller/add-chain-controller.ts

```diff
@@ -0,0 +1,188 @@
+// Copyright 2020-2022 OnFinality Limited authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import * as fs from 'fs';
+import * as path from 'path';
+import {ProjectManifestParentV1_0_0, getProjectRootAndManifest, getSchemaPath} from '@subql/common';
+import * as yaml from 'js-yaml';
+
+interface DockerComposeType {
+  version: string;
+  services: any;
+}
+
+export function addChain(multichain: string, chainManifestPath: string, chainId: string, schema: string) {
+  const multichainManifestPath = determineMultichainManifestPath(multichain);
+  const multichainManifest = loadOrCreateMultichainManifest(multichainManifestPath);
+  chainManifestPath = handleChainManifestOrId(chainManifestPath, chainId, schema, multichainManifestPath);
+  validateAndAddChainManifest(chainManifestPath, multichainManifest);
+  fs.writeFileSync(multichainManifestPath, yaml.dump(multichainManifest));
+  updateDockerCompose(chainManifestPath);
+}
+
+function determineMultichainManifestPath(multichain: string): string {
+  // Default multichain manifest path
+  const defaultMultichainManifestPath = path.resolve(process.cwd(), 'multichain-manifest.yaml');
+
+  return multichain ? path.resolve(multichain) : defaultMultichainManifestPath;
+}
+
+function loadOrCreateMultichainManifest(multichainManifestPath: string): ProjectManifestParentV1_0_0 {
+  let multichainManifest: ProjectManifestParentV1_0_0;
+
+  if (fs.existsSync(multichainManifestPath)) {
+    multichainManifest = yaml.load(fs.readFileSync(multichainManifestPath, 'utf8')) as ProjectManifestParentV1_0_0;
```

Comment:
I think we had a new way of loading and writing yaml that retained the comments in the file